This patch is only needed for old kernels that do not have SOCK_CLOEXEC flag - used in tomato and keenetic
--- a/src/fdevent.c
+++ b/src/fdevent.c
@@ -21,12 +21,22 @@
 #include <share.h>      /* _SH_DENYRW */
 #endif
 
+#ifndef LINUX_VERSION_CODE
+#include <linux/version.h>
+#endif
+
+/* SOCK_CLOEXEC & SOCK_NONBLOCK since Linux 2.6.27 */
+#if LINUX_VERSION_CODE > KERNEL_VERSION(2,6,36)
 #ifdef SOCK_CLOEXEC
 static int use_sock_cloexec;
 #endif
 #ifdef SOCK_NONBLOCK
 static int use_sock_nonblock;
 #endif
+#else
+#undef SOCK_CLOEXEC
+#undef SOCK_NONBLOCK
+#endif /* SOCK_CLOEXEC & SOCK_NONBLOCK since Linux 2.6.27 */
 
 int fdevent_config(const char **event_handler_name, log_error_st *errh) {
 	static const struct ev_map { fdevent_handler_t et; const char *name; } event_handlers[] =
